integrations/huggingface-transformers/tutorials/question-answering/question_answering-cli.md

```diff
@@ -0,0 +1,372 @@
+
+Run the following to inspect the Rotten Tomatoes dataset.
+
+```python
+from datasets import load_dataset
+
+rotten_tomatoes = load_dataset("rotten_tomatoes")
+print(rotten_tomatoes)
+print(rotten_tomatoes["train"][0])
+```
+
+Output:
+```bash
+{'answers': {'answer_start': [199, 196, 199],
+             'text': ['cleans the entire floor',
+                      'it cleans the entire floor.',
+                      'cleans the entire floor']},
+ 'context': 'We are in love with our Roomba. The Roomba does as is promised. '
+            'Much like all of the reviews we read before purchasing, we '
+            'experience that the robot takes its own path appearing erratic; '
+            'however, it cleans the entire floor. For large rooms (like our '
+            'open floor plan living / dining kitchen) you should definitely '
+            'use the included virtual wall. It works well. The first 3-4 times '
+            'we used it, we just watched in awe as the Roomba cleaned our '
+            'floor. You should definitely clean the brushes once a week or so, '
+            "but that's an easy task compared to sweeping entire rooms. Roomba "
+            "will also easily encourage you to keep your rooms tidy. it's much "
+            'easier to concentrate on keeping big items off the floor than '
+            'sweeping the entire floor daily. Roomba has taught us tidier '
+            'habits and has keep our hardwoods grime free. We have several '
+            'friends who have their Roombas on order!',
+ 'id': '5dd4980bcc027a086d65df4a',
+ 'question': 'What does it do?',
+ 'title': 'Amazon_Reviews_330'}
+```
+
+We can see that each row dataset contains the following:
+- A `context` field which is a string representing the text which contains the answer
+- A `question` field which is a string representing the query
+- An `answers` dictionary, which contains a `answers_start` (a list of ints) and `text` (a list of strings). `text` is the raw strings that are the correct answers
+and `answer_start` are the index of the first character in the `context`. For the example above, the `c` in `cleans the entire floor` is the 199th character of `context`.
+
+The `question_answering` training script accepts JSON files in the form:
+
+### Using Local JSON Files
+
+Let's walk through how to pass this dataset in JSON dataset to the CLI.
+
+#### Save Dataset as a JSON File
+
+The `question_answering` training script accepts JSON files in the form:
+
+```bash
+{
+  'data': [
+    {'question': 'What is my Name?', 'context': 'My name is Robert', "answers":{'answer_start':[11], 'text':['Robert']}},
+    {'question': 'What is my Name?', 'context': 'My name is Mark', "answers":{'answer_start':[11], 'text':['Mark']}},
+    {'question': 'What is my Name?', 'context': 'My name is Ben', "answers":{'answer_start':[11], 'text':['Ben']}},
+    ...
+  ]
+}
+```
+
+Run the following to convert the dataset to this format and dump to a json file.
```

Comment:
Run the following to convert the dataset to this format and dump to a `json` file.